File: 351002/Valinski/lab4/KafkaTraining/Publisher/Publisher.Domain/Models/Topic.cs
namespace Publisher.Domain.Models;

public class Topic
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public virtual User User { get; set; } = null!;
    
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }

    public virtual ICollection<TopicLabel> TopicLabels { get; set; } = new List<TopicLabel>();
}

File: 351004/Lavrenov/Lab_1/services/topic.py
from models.topic import Topic
from dto.requests import TopicRequestTo
from dto.responses import TopicResponseTo
from .base import BaseService


class TopicService(BaseService[Topic, TopicRequestTo, TopicResponseTo]):
    def _to_entity(self, request: TopicRequestTo) -> Topic:
        return Topic(
            userId=request.userId,
            title=request.title,
            content=request.content,
            markerIds=request.markerIds or [],
        )

    def _to_response(self, entity: Topic) -> TopicResponseTo:
        return TopicResponseTo(
            id=entity.id,
            userId=entity.userId,
            title=entity.title,
            content=entity.content,
            markerIds=entity.markerIds,
            created=entity.created,
            modified=entity.modified,
        )
